lib/web_ui/dev/generate_builder_json.dart

```diff
@@ -110,71 +110,62 @@ Map<String, dynamic> _getBundleBuildStep(TestBundle bundle) {
+Map<String, dynamic> _getTestStepForPlatformAndBrowser(
   List<TestSuite> suites,
   String platform,
-  bool Function(TestSuite suite) filter, {
+  BrowserName browser, {
   String? specificOS,
   String? cpu,
 }) {
-  return suites
-    .where(filter)
-    .map((TestSuite suite) => <String, dynamic>{
-        'name': '$platform run ${suite.name} suite',
-        'recipe': 'engine_v2/tester_engine',
-        'drone_dimensions': <String>[
-          'device_type=none',
-          'os=${specificOS ?? platform}',
-          if (cpu != null) 'cpu=$cpu',
-        ],
-        'gclient_variables': <String, dynamic>{
-          'download_android_deps': false,
-          'download_jdk': false,
+  final filteredSuites = suites.where((suite) => suite.runConfig.browser == browser);
+  final bundles = filteredSuites.map((suite) => suite.testBundle).toSet();
+  return <String, dynamic>{
+    'name': '$platform run ${browser.name} suites',
+    'recipe': 'engine_v2/tester_engine',
+    'drone_dimensions': <String>[
+      'device_type=none',
+      'os=${specificOS ?? platform}',
+      if (cpu != null) 'cpu=$cpu',
+    ],
+    'gclient_variables': <String, dynamic>{
+      'download_android_deps': false,
+      'download_jdk': false,
+    },
+    'dependencies': <String>[
+      'web_tests/artifacts',
+      ...bundles.map((bundle) => 'web_tests/test_bundles/${bundle.name}'),
+    ],
+    'test_dependencies': <dynamic>[
+      <String, dynamic>{
+        'dependency': 'goldctl',
+        'version': 'git_revision:720a542f6fe4f92922c3b8f0fdcc4d2ac6bb83cd',
+      },
+      if (browser == BrowserName.chrome)
+        <String, dynamic>{
+          'dependency': 'chrome_and_driver',
+          'version': '125.0.6422.141',
         },
-        'dependencies': <String>[
-          'web_tests/artifacts',
-          'web_tests/test_bundles/${suite.testBundle.name}',
-        ],
-        'test_dependencies': <dynamic>[
-          <String, dynamic>{
-            'dependency': 'goldctl',
-            'version': 'git_revision:720a542f6fe4f92922c3b8f0fdcc4d2ac6bb83cd',
-          },
-          if (suite.runConfig.browser == BrowserName.chrome)
-            <String, dynamic>{
-              'dependency': 'chrome_and_driver',
-              'version': '125.0.6422.141',
-            },
-          if (suite.runConfig.browser == BrowserName.firefox)
-            <String, dynamic>{
-              'dependency': 'firefox',
-              'version': 'version:106.0',
-            }
-        ],
-        'tasks': <dynamic>[
-          <String, dynamic>{
-            'name': 'run suite ${suite.name}',
-            'parameters': <String>[
-              'test',
-              '--run',
-              '--suite=${suite.name}'
-            ],
-            'script': 'flutter/lib/web_ui/dev/felt',
-          }
-        ]
-      }
-    );
+      if (browser == BrowserName.firefox)
+        <String, dynamic>{
+          'dependency': 'firefox',
+          'version': 'version:106.0',
```

Comment:
I find it a little scary to hard code dependency versions in code. Is it documented that when we upgrade these dependencies that we should also change the code on these lines?

Reply:
Yeah actually I agree. Maybe I can parse these out of the `browser_lock.yaml` file or whatever.